pycytominer/cyto_utils/cells.py

```diff
@@ -410,9 +438,30 @@ def load_compartment(self, compartment):
         pandas.core.frame.DataFrame
             Compartment dataframe.
         """
-        compartment_query = "select * from {}".format(compartment)
-        df = pd.read_sql(sql=compartment_query, con=self.conn)
-        return df
+
+        # Get data useful to pre-alloc memory
+        num_cells = self.count(compartment)
+        meta_cols, feat_cols = self.get_columns(compartment)
+        num_meta, num_feats = len(meta_cols), len(feat_cols)
+
+        # Use pre-allocated np.array for data
+        feats = np.empty(shape=(num_cells, num_feats), dtype=np.float64)
+        # Use pre-allocated pd.DataFrame for metadata
+        metas = pd.DataFrame(columns=meta_cols, index=range(num_cells))
+
+        # Query database for selected columns of chosen compartment
+        columns = ", ".join(meta_cols + feat_cols)
+        query = f"select {columns} from {compartment}"
+        resultset = self.conn.execute(query)
+
+        # Load data row by row for both meta information and features
+        for i, row in enumerate(resultset):
+            metas.loc[i] = row[:num_meta]
+            feats[i] = row[num_meta:]
+
+        # Return concatenated data and metainformation of compartment
+        return pd.concat(
```

Comment:
metadata should be indexed first, right? Am I reading correctly that it is indexed _after_ morphology features in this setup?

Reply:
Ah, we can equally write:

```
return pd.concat([metas, pd.DataFrame(columns=feat_cols, data=feats)], axis=1)
```

Is this preferred?

Reply:
We are only using indexing via `:num_meta` or `num_meta:` for the SQL query result.

Reply:
The standard is to have metadata features before morphology features in profile data.

>  Ah, we can equally write:
> return pd.concat([metas, pd.DataFrame(columns=feat_cols, data=feats)], axis=1)
> Is this preferred?


If I am correct in my read of the `pd.concat()` call that metadata will be added as the last columns, then yes please.

Reply:
Just to clarify. My read of `pd.concat()` is incorrect? Metadata features are inserted as the first columns?

Reply:
No, currently not. Because the existing pytests assume that the structure is morphology features, then meta features.

Reply:
Interesting! I added #220 (this fix is beyond scope of your PR)

Reply:
I changed the existing pytests to agree with the order meta features, then morphological features, and swapped the order of the concat.

Reply:
wonderful!